src/JSON-RPC/specification.md

```diff
@@ -0,0 +1,505 @@
+# Ethereum Execution Layer JSON-RPC API Specification
+## Working Draft: Updated December 9th  
+---
+### **Author:**
+Jared Doro(jareddoro@gmail.com)
+
+### **Editors:**
+
+
+### **Abstract:**
+This document provides a detailed description of Ethereum's Execution Layer API. This document also provides the minimum requirements and functionality that is needed for a piece of software to be considered a valid Ethereum Execution Layer API.
+### **Keywords:**
+The keywords **MUST**, **MUST NOT**, **REQUIRED**, **SHALL**, **SHALL NOT**, **SHOULD**, **SHOULD NOT**, **RECOMMENDED**, **NOT RECOMMENDED**, **MAY**, and **OPTIONAL** in this document are to be interpreted as described in [[RFC2119](http://www.ietf.org/rfc/rfc2119.txt)] when, and only when, they appear in all capitals, as shown here.
+
+# 1 Introduction 
+
+The Ethereum execution layer API is one of the key components of Ethereum. It acts as an intermediary between the users of Ethereum and the execution layer where the transactions are received and executed. It provides a way for users to send transactions, request data, and execute smart contracts. 
+## 1.1 Purpose and Intended Audience
+
+The purpose of this document is to act as a centralized source of information regarding the functional and non-functional requirements for Ethereum's execution layer API. This document is intended for development teams that are planning on implementing a version of the execution layer API. This document would also be beneficial to but is not intended for anyone interested in learning how the user interacts with Ethereum clients and at the most basic level. 
+
+## 1.2 Scope
+ 
+The Ethereum execution layer API provides the basis for all external interactions with the Ethereum blockchain. The interactions with the Ethereum network can be divided into four categories. Transferring ETH, deploy contracts, executing contracts, and administrative tasks. The API also provides endpoints that returns historical network and block data. 
+
+## 1.3 Definitions and Terms
+* ETH
+* Smart contracts
+* user
+* client
+* EVM
+* Account
+* block
+* block-chain
+
+### 1.3.1 Not Specified vs Null
```

Comment:
Recommend "Optional" rather than "Not specified" for referring to properties that can be left out.